### ISSUE TITLE
Pages not showing content after editing in frontend

### ISSUE THREAD

**Issue by kevinandrewsde** [FriendsOfTYPO3/frontend_editing#402]
We are having an issue that popped up after the 1.9.0 release in which we sometimes get a PHP exception thrown on page loading of pages that were opened in frontend editing, at the following line:

https://github.com/FriendsOfTYPO3/frontend_editing/blob/335c06f0946f0706ae8acb040fdf92b0991e6c7b/Classes/Hook/FrontendEditingInitializationHook.php#L172

When the error occurs the $parentObject->content is not set which causes an exception. Weirdly, even clearing cache etc will not fix the problem, from that point on, the page is only viewable when logged in to the backend (i.e. while frontend editing), and completely broken for site visitors, as every call causes the exception. The fix is to deactivate the frontend editing extension, then clear cache and reload the page in question, which works every time.

Rolling back to 1.8.3 fixes this issue entirely it seems. I am sorry, but i cant offer any further input as it does not seem as if the Line referenced above is the root cause.

**Comment by mabolek** [comment]
@kevinandrewsde Thank you for this. It looks like the solution would be to check if `$parentObject->content` is set. Which TYPO3 version are you using?

**Comment by kevinandrewsde** [comment]
I am using the latest LTS version from composer (9.5.13). Currently rolled back to frontend_editing 1.8.3 though until this is resolved.

**Comment by kevinandrewsde** [comment]
> @kevinandrewsde Thank you for this. It looks like the solution would be to check if `$parentObject->content` is set. Which TYPO3 version are you using?

I have tried this fix locally, i am closer now to discovering the root cause: the $parentObject actually has no content, and the reason seems to be that frontend editing puts a broken version of the page into the cache. When navigating the frontend with ?no_cache=1 everything works as intended. That is also why the pages display correctly when logged in and using the frontend editing extension, as, to my knowledge, the extension disables cache while it is active.

**Comment by kevinandrewsde** [comment]
I think i might have identified the issue (see above references)
I cant repro the issue after reverting those two lines

**Comment by mabolek** [comment]
Hi @kevinandrewsde! 

That's odd, as it `GeneralUtility::getUserObj()` is deprecated in TYPO3 v9 and should be replaced with `GeneralUtility::makeInstance()`. The two calls should be identical.

Regarding caches, there is a bug in how no_cache is handled in v9. I think it's unrelated, but it has affected other things in the Frontend Editing extension. https://forge.typo3.org/issues/87227/

**Comment by kevinandrewsde** [comment]
Ok forget my previous comments about the GeneralUtility Lines. I have many different TYPO3 Sites and I made changes to the PHP file on one while testing another...fail.
I finally figured out that causes this bug for me: some of my sites had the following as part of their template. 
[globalVar = TSFE : beUserLogin > 0] 
[global]
These sites work fine with 1.9.1. Sites that do not have these directives in their main template do not work. I found a change in which you removed exactly those two lines from your setup.ts 899534e. I assume thats whats causing the issue.

The workaround for me is to simply add those lines to all my maintemplates for now.
